### PR TITLE
Staging branch for next major release 3.0.0

### PR DESCRIPTION
**Pull request by evan-masseau**
Individual PRs to this branch should get reviewed like they were going in to master, and we will conduct final review of the whole set of changes together prior to 3.0.0 release.

### CODE REVIEW

File: sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt [hunk 1]
```diff
@@ -83,14 +84,21 @@ internal object KlaviyoApiClient : ApiClient {
...
         enqueueRequest(ProfileApiRequest(profile))
     }
 
     override fun enqueuePushToken(token: String, profile: Profile) {
+        Registry.log.verbose("Enqueuing Push Token request")
         enqueueRequest(PushTokenApiRequest(token, profile))
     }
 
+    override fun enqueueUnregisterPushToken(apiKey: String, token: String, profile: Profile) {
+        enqueueRequest(UnregisterPushTokenApiRequest(apiKey, token, profile))
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt:97]:
```suggestion
        Registry.log.verbose("Enqueuing Unregister Push Token  request")
        enqueueRequest(UnregisterPushTokenApiRequest(apiKey, token, profile))
```

File: gradle.properties [hunk 2]
```diff
@@ -26,5 +26,8 @@ android.enableJetifier=true
 # URL to Klaviyo server
 klaviyoServerUrl=https://a.klaviyo.com
 
+# Klaviyo API Revision
+klaviyoApiRevision=2023-07-15
```
Comment (dan-peluso) [inline gradle.properties:30]: This is referring to a backend version, right?

Comment (evan-masseau) [reply gradle.properties:30]: Correct

**Review by dan-peluso — APPROVED** [review]

File: MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -2,6 +2,19 @@
...
+# 3.0.0
+
+### Improvements
+- The Klaviyo Android SDK now automatically tracks changes to the
+  user's notification permission whenever the app is opened or resumed.
+- Additionally, the SDK will now hold the push token internally after you `resetProfile`
+  and automatically attach the token to the next profile. This is a change from past behavior where the token
+  would need to be explicitly set again after resetting.   
+
+### Removals
```
[the remark this answers is not in the document]
Comment (dan-peluso) [reply MIGRATION_GUIDE.md:14]: I'm down for that

[the remark this answers is not in the document]
Comment (dan-peluso) [reply MIGRATION_GUIDE.md:14]: https://github.com/klaviyo/klaviyo-android-sdk/pull/193

**Review by dan-peluso — APPROVED** [review]
LGTM

**Review by kennyklaviyo — APPROVED** [review]

**Pull request merged**